File: app/src/main/java/com/example/mg/todo/data/model/NoteModel.java
package com.example.mg.todo.data.model;

import android.os.Parcel;
import android.os.Parcelable;

public class NoteModel implements Parcelable {
    private String mText;
    private String mDescription;
    private String mImage;
    private String mDate;

    public static final Creator<NoteModel> CREATOR = new Creator<NoteModel>() {
        @Override
        public NoteModel createFromParcel(Parcel in) {
            return new NoteModel(in);
        }

        @Override
        public NoteModel[] newArray(int size) {
            return new NoteModel[size];
        }
    };

    public NoteModel() {

    }

    private NoteModel(Parcel in) {
        mText = in.readString();
        mDescription = in.readString();
        mImage = in.readString();
        mDate = in.readString();

    }

    public String getText() {
        return mText;
    }

    public void setText(String text) {
        this.mText = text;
    }

    public String getDescription() {
        return mDescription;
    }

    public void setDescription(String description) {
        this.mDescription = description;
    }

    public String getImage() {
        return mImage;
    }

    public void setImage(String mImage) {
        this.mImage = mImage;
    }

    public String getmDate() {
        return mDate;
    }

    public void setmDate(String mDate) {
        this.mDate = mDate;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(mText);
        parcel.writeString(mDescription);
        parcel.writeString(mImage);
        parcel.writeString(mDate);
    }


}

File: app/src/main/java/com/example/mg/todo/data/model/NoteModel.kt
package com.example.mg.todo.data.model

import android.os.Parcel
import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "notes")
open class NoteModel(
        var text: String? = "",
        var description: String? = "",
        var image: String? = null,
        var date: String? = ""
) : Parcelable {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
        id = parcel.readLong()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(text)
        parcel.writeString(description)
        parcel.writeString(image)
        parcel.writeString(date)
        parcel.writeLong(id)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<NoteModel> {
        override fun createFromParcel(parcel: Parcel): NoteModel {
            return NoteModel(parcel)
        }

        override fun newArray(size: Int): Array<NoteModel?> {
            return arrayOfNulls(size)
        }
    }


}